product-service/handlers/createProduct.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const {Client} = require('pg');
+
+const {PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD} = process.env;
+
+const dbOptions = {
+    host: PG_HOST,
+    port: PG_PORT,
+    database: PG_DATABASE,
+    user: PG_USERNAME,
+    password: PG_PASSWORD,
+    ssl: {
+        rejectUnauthorized: false
+    },
+    _connectionTimeoutMillis: 5000
+}
+
+const validHeaders = {
+    'Access-Control-Allow-Origin': '*',
+    'Access-Control-Allow-Credentials': true,
+}
+
+export const createProduct = async (event) => {
+    const client = new Client(dbOptions);
+
+    // Log incoming event
+    console.log('event: ', event);
+
+    let body = event.body;
+
+    let {
+        title = '',
+        description = '',
+        price = '',
+        count = 0
+    } = body;
+
+    // Check if product data is invalid
+    if (title === '' || description === '' || price === '' || count === '') {
+        return {
+            statusCode: 400,
+            headers: validHeaders,
+            body: JSON.stringify({
+                error: 'Wrong post data'
+            }),
+        };
+    }
+
+    // Connect to database
+    try {
+        await client.connect();
+        await client.query("BEGIN"); // Start transaction
+    } catch (err) {
+        return {
+            statusCode: 500,
+            headers: validHeaders,
+            body: JSON.stringify({
+                error: 'Cannot connect to database:' + err
+            }),
+        };
+    }
+
+    // Insert
+    try {
+        const newProductResult = await client.query(
```

Comment:
I would suggest to move all query logic to some separate module and create a product service that would handle data base connections and queries